src/main/java/org/mule/extension/http/internal/request/HttpRequestOperations.java

```diff
@@ -244,10 +254,37 @@ protected String buildPath(String basePath, String path) {
   public void initialise() throws InitialisationException {
     defaultStatusCodeValidator = new SuccessStatusCodeValidator("0..399");
     defaultRequestBuilder = new HttpRequesterRequestBuilder();
-    httpRequester = new HttpRequester(new HttpRequestFactory(), new HttpResponseToResult(), new HttpErrorMessageGenerator());
+    // Profiling API is only available for use in 4.5.0 or if enabled
+    httpResponseProfilingEnabled = runtimeVersion.atLeast("4.5.0-SNAPSHOT") || getBoolean(HTTP_ENABLE_PROFILING);
```

Comment:
do you really want this enabled by default on 4.5?

Reply:
Available only by system property. Probably this should be changed in the future (best case feature flagging/toggling or activation strategy for extensions).